File: trunk/tinyember/TinyEmberPlusRouter/TinyEmberPlusRouter/model/StringParameter.cpp
#include "StringParameter.h"
#include "ElementVisitor.h"

namespace model
{
   StringParameter::StringParameter(int number, Element* parent, std::string const& identifier, NotificationSink* notificationSink)
      : Parameter(number, parent, identifier, notificationSink)
   {}

   void StringParameter::accept(ElementVisitor* visitor)
   {
      visitor->visit(this);
   }

   void StringParameter::onValueChanged()
   {
      auto notificationSink = this->notificationSink();

      if(notificationSink != nullptr)
         notificationSink->notifyParameterValueChanged(path(), value());
   }
}

File: tags/emberplussdk-1-6-0/EmberLib.net/EmberPlusRouter/Model/StringParameter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace EmberPlusRouter.Model
{
   class StringParameter : Parameter<string>
   {
      public StringParameter(int number, Element parent, string identifier, Dispatcher dispatcher, bool isWriteable)
      : base(number, parent, identifier, dispatcher, isWriteable)
      {
      }

      public override TResult Accept<TState, TResult>(IElementVisitor<TState, TResult> visitor, TState state)
      {
         return visitor.Visit(this, state);
      }
   }
}
